mitosheet/src/components/modals/ReplayAnalysisModals.tsx

```diff
@@ -0,0 +1,145 @@
+            viewComponent={
+                <Fragment>
+                    
+                    <div className='text-align-left text-body-1'>
+                        The analysis_to_replay {props.analysisName} does not exist on this computer. TODO: what should we say here? Should we overwrite the analysis ID in the calling cell? But thats confusing.              
+                    </div>
+                </Fragment>
+            }
+            buttons={
+                <>
+                    <TextButton
+                        variant='light'
+                        width='small'
+                        onClick={() => {props.setUIState((prevUIState) => {
+                            return {
+                                ...prevUIState,
+                                currOpenModal: {type: ModalEnum.None}
+                            }
+                        })}}
+                    >
+                        Close
+                    </TextButton>
+                    <TextButton
+                        variant='dark'
+                        width='medium'
+                        href={DISCORD_INVITE_LINK}
+                        target='_blank'
+                        onClick={() => {
+                            props.setUIState((prevUIState) => {
+                                return {
+                                    ...prevUIState,
+                                    currOpenModal: {type: ModalEnum.None}
+                                }
+                            })
+                            // TODO: do we want to change the parameter here, so they don't get overwritten?
+                            return true;
+                        }}
+                    >
+                        Get Immediate Support
+                    </TextButton>
+                </> 
+            }
+        />
+    )    
+};
+
+/*
+    This modal displays to the user when the analysis that they are 
+    replaying fails to run correctly.
+*/
+export const InvalidReplayedAnalysisModal = (
+    props: {
+        error: MitoError | undefined, 
+        setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+        mitoAPI: MitoAPI;
+        oldAnalysisName: string;
+        newAnalysisName: string;
+    }): JSX.Element => {
+    const [viewTraceback, setViewTraceback] = useState(false);
+
+    if (props.error === undefined) {
+        return (<></>)
+    }
+
+    return (
+        <DefaultModal
+            header={"Analyis Could Not Be Replayed"}
+            modalType={ModalEnum.Error}
+            wide
+            viewComponent={
+                <Fragment>
+                    {props.error.to_fix &&
+                        <div className='text-align-left text-body-1' onClick={() => setViewTraceback((viewTraceback) => !viewTraceback)}>
+                            There was an error in replaying your analysis. The generated code can be found below, but this anlaysis could not be replayed on the new dataset. {' '}
```

Comment:
Maybe we could change this to: "There was an error replaying your analysis because of changes in the input data. You can see the previous analysis in the code cell below."

I wish there was a way that we didn't have to just offer the user an option to start a new analysis. Because it means that the user can't correct their mistake. I want to figure out how we can give users a way to fix their mistake and then continue working in the same analysis. 

Maybe a second option could be "close mitosheet" which doesn't update the analysisID and closes the Mitosheet so that the user can correct the changes that they've made to the dataframes and then try again.

Reply:
I took your language too.

I wish we could do something different too - but given that this specification and work is about handling generated code getting deleted better and not about making replaying analyses error handling better - let's leave as is.

There's an open issue for it here #180  - I think we can do _much_ better!